work/v1/types.go

```diff
@@ -52,14 +52,16 @@ type ManifestWork struct {
 // ManifestWorkSpec represents a desired configuration of manifests to be deployed on the spoke cluster.
 type ManifestWorkSpec struct {
 	// Workload represents the manifest workload to be deployed on spoke cluster
-	Workload ManifestsTemplate `json:"workload,omitempty" protobuf:"bytes,1,opt,name=workload"`
+	Workload []ManifestsTemplate `json:"workload,omitempty" protobuf:"bytes,1,opt,name=workload"`
 }
 
 // ManifestsTemplate represents the manifest workload to be deployed on spoke cluster
 type ManifestsTemplate struct {
-	// Manifests represents a list of kuberenetes resources to be deployed on the spoke cluster.
-	// +optional
-	Manifests []runtime.RawExtension `json:"manifests,omitempty" protobuf:"bytes,2,rep,name=manifests"`
```

Comment:
@sttts can we keep this a slice and have each member of the slice unchecked?

Reply:
I believe you have to type alias `runtime.RawExtension` with controller-tools and annotate the new type with `+kubebuilder:validation:EmbeddedResource` and `+kubebuilder:pruning:PreserveUnknownFields`.

Reply:
```Golang
Manifests []Manifest `json:"manifests,omitempty" protobuf:"bytes,2,rep,name=manifests"`

// +kubebuilder:validation:EmbeddedResource
// +kubebuilder:pruning:PreserveUnknownFields
type Manifest runtime.RawExtension
```

Reply:
it seems not work, the generated crd has a field like
```
manifest:
     description: Manifests represents a list of kuberenetes resources
          to be deployed on the spoke cluster.
      type: array
      items:
         $ref: '#/definitions/k8s.io~1apimachinery~1pkg~1runtime~0RawExtension'
```

Reply:
inlining RawExtension works. Please help to check if it is a valid pattern.